### DISCUSSION TITLE
Multiple coordinates for same dimension?

### DISCUSSION THREAD

**Discussion by RubendeBruin** [pydata/xarray#5482] [Q&A] [answered]
I'm currently writing some code that sits between two existing packages. The code deals with frequency domain spectrums. The problem that I have is that the first package defines the frequency in [radians per second] while the other package uses [Hz].

My data has a dimension "frequency". I'm looking for a way to expose it in a way that makes it easy to use from either package. I think it would be convenient to somehow attach two sets of labels to the frequency axis, one being in Hz and the other one being the values expressed in rad/s.

I can see similar use-cases for people working with the same data expressed in different units such as ft and m or F and C.

Is this already possible? Or does anyone have a good suggestion on how to implement this?

**Comment by keewis** [comment, marked as answer]
It should be possible to use non-dimension coordinates but you would need to use distinct names like "frequency" and "angular_velocity". For example:
```python
ds = (
    xr.Dataset(coords={"frequency": [1, 2, 3]})
    .assign_coords(angular_velocity=lambda ds: ("frequency", ds.frequency * 2 * np.pi))
)
```
---
I would also recommend to have a look at [`pint-xarray`](https://github.com/xarray-contrib/pint-xarray) which should make working with units easier. It is still experimental, though, so I very much welcome feedback.

**Comment by AlexAtCCRI** [comment]
Is there a way to have multiple coordinates for a single dimension which can all be sliced? 
like `x_ft`, `x_m`,?

**Comment by niksirbi** [comment]
I'm also interested in @AlexAtCCRI 's question and specifically the following case:

Can we have a dimension with two coordinates - one "dimension coordinate" and one "non-dimension coordinate" - both of which are indexed (i.e. can be used for slicing and alignment)?

I use the above terms as defined [here](https://docs.xarray.dev/en/latest/user-guide/terminology.html).

**Reply by keewis** [reply]
it's been a while since discussion was opened initially, and a lot has changed. By now you can have multiple indexed coordinates along the same dimension (use `set_xindex` / `drop_indexes`). This only helps for indexing (slicing), though, you still can only have one single index along a dimension for alignment.

If the aim of this is to deal with different units, I recommend having a look at `ds.pint.sel(x=ureg.Quantity(..., "ft"))` (from `pint-xarray`), which will do the right thing without the need to duplicate coordinates.

**Reply by niksirbi** [reply]
Fantastic! I'll definitely look into pint, it sounds like it could cover our use-case.
